cogeo_mosaic/backends/sqlite.py

```diff
@@ -0,0 +1,314 @@
+"""cogeo-mosaic SQLite backend."""
+
+import itertools
+import json
+import re
+import sqlite3
+from typing import Dict, List, Sequence
+from urllib.parse import urlparse
+
+import attr
+import mercantile
+from cachetools import TTLCache, cached
+from cachetools.keys import hashkey
+
+from cogeo_mosaic.backends.base import BaseBackend
+from cogeo_mosaic.backends.utils import find_quadkeys
+from cogeo_mosaic.cache import cache_config
+from cogeo_mosaic.errors import MosaicExistsError, MosaicNotFoundError
+from cogeo_mosaic.logger import logger
+from cogeo_mosaic.mosaic import MosaicJSON
+from cogeo_mosaic.utils import bbox_union
+
+sqlite3.register_adapter(dict, json.dumps)
+sqlite3.register_adapter(tuple, json.dumps)
+sqlite3.register_adapter(list, json.dumps)
+sqlite3.register_converter("JSON", json.loads)
+
+
+@attr.s
+class SQLiteBackend(BaseBackend):
+    """SQLite Backend Adapter."""
+
+    db_path: str = attr.ib(init=False)
+    mosaic_name: str = attr.ib(init=False)
+
+    db: sqlite3.Connection = attr.ib(init=False)
+
+    _backend_name = "SQLite"
+    _metadata_table: str = "mosaicjson_metadata"
+
+    def __attrs_post_init__(self):
+        """Post Init: parse path connect to Table.
+
+        A path looks like
+
+        sqlite:///{db_path}:{mosaic_name}
+
+        """
+        if not re.match(r"^sqlite:///.+\:[a-zA-Z0-9\_\-\.]+$", self.path,):
+            raise ValueError(f"Invalid SQLite path: {self.path}")
+
+        parsed = urlparse(self.path)
+        path = parsed.path.lstrip("/")
+
+        self.mosaic_name = path.split(":")[-1]
+        assert (
+            self.mosaic_name is not self._metadata_table
+        ), f"'{self._metadata_table}' is a reserved table name."
+
+        self.db_path = path.replace(f":{self.mosaic_name}", "")
+
+        self.db = sqlite3.connect(self.db_path, detect_types=sqlite3.PARSE_DECLTYPES)
+        self.db.row_factory = sqlite3.Row
+
+        # Here we make sure the mosaicJSON.name is the same
+        if self.mosaic_def:
+            self.mosaic_def.name = self.mosaic_name
```

Comment:
because we use the `name` later in the SQL query we have to make sure the `name` is the same as the name provided in the uri

Reply:
Might be good to add a warning here. Users might be surprised name changed?